File: aat-android/src/main/java/ch/bailu/aat/description/FloatDescription.java
package ch.bailu.aat.description;

import ch.bailu.aat_lib.description.ContentDescription;


public abstract class FloatDescription extends ContentDescription {

    private float cache=0f;

    protected float getCache() {
        return cache;
    }

    protected boolean setCache(float f) {
        boolean r;
        if (cache==f) {
            r=false;
        } else {
            r=true;
            cache=f;
        }
        return r;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/FloatDescription.kt
package ch.bailu.aat_lib.description

abstract class FloatDescription : ContentDescription() {
    protected var cache = 0f
        private set

    protected fun setCache(f: Float): Boolean {
        val r: Boolean
        if (cache == f) {
            r = false
        } else {
            r = true
            cache = f
        }
        return r
    }
}
